### ISSUE TITLE
Gnome language & locale settings are not applied

### ISSUE THREAD

**Issue by demokritos** [NixOS/nixpkgs#166913]
### Describe the bug
I changed gnome language settings and re-logged in, but language and locale didn't change.
It seems that locale keeps to `i18n.defaultLocale`. `locale` command in a terminal prints only with that default locale.

### Steps To Reproduce
1. set i18n.supportedLocales to an array of multiple locales in `/etc/nixos/configuration.nix`
2. nixos-rebuild switch
3. change language to one of the supported locales in gnome settings
4. re-log in

### Expected behavior
Interface language should be changed

### Screenshots
![스크린샷, 2022-04-02 17-47-08](https://user-images.githubusercontent.com/3399860/161375273-44dc5051-b120-450d-9391-f92fde103b17.png)
Language setting is English, but the actual interface is in Korean. 

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.0-zen1, NixOS, 22.05 (Quokka), 22.05pre365028.710fed5a248`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - channels(root): `"nixos"`
 - channels(esrevinu): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by jtojnar** [comment]
GNOME Settings [uses systemd-localed](https://discourse.gnome.org/t/is-it-possible-to-easily-create-a-custom-system-format-if-not-such-a-feature-would-be-a-great-addition/6739/2) to change locale but that does not work on NixOS, since `/etc/locale.conf` is already [managed by the NixOS module](https://github.com/NixOS/nixpkgs/blob/35de448c4530c562bfd0d1c2a2d0bf76d1d4e468/nixos/modules/config/i18n.nix#L89-L94):

```ShellSession
$ localectl set-locale LANG=en_UK.UTF-8
Failed to issue method call: Changing system settings via systemd is not supported on NixOS.
```

Additionally, changing the language in Control Center successfully sets the language for your user in AccountsService but I have no idea if anything actually uses that:

```ShellSession
$ gdbus call --system -d org.freedesktop.Accounts -o /org/freedesktop/Accounts/User1000 -m org.freedesktop.Accounts.User.SetLanguage "'en_UK.UTF-8'"
()
$ sudo cat /var/lib/AccountsService/users/jtojnar
[User]
Language=en_UK.UTF-8
…
```

We probably need to remove all the panels that are not supported, or replace them with instructions how to do those things declaratively.

**Comment by SamLukeYes** [comment]
Switching the login shell to xonsh can work around this issue. There's probably some problem with xonsh, which causes https://github.com/NixOS/nixpkgs/issues/188404, but it happens to behave as expected in this situation.

**Comment by vlinkz** [comment]
Ok did a bit of digging, and think I found what's causing the issue. It seems that due to #48517, the environment variables declared in the NixOS configuration override any variables set by GNOME locally. #53448 and #48255 seem to be related.

**Comment by me00001** [comment]
Does this issue was exist in 22.05 ?

**Comment by 2xsaiko** [comment]
> We probably need to remove all the panels that are not supported, or replace them with instructions how to do those things declaratively.

No, these settings should work. Not every NixOS deployment is a single-user system or system where every user can modify the configuration where that is reasonable. For example, in a company-wide setup especially with users served by LDAP or something like that, that would be a nightmare to maintain.
